### PR TITLE
Key undefined

### LINKED ISSUE #237: `key` is undefined
See: %h4J9aYMcmT68dtiZNAP6pS0LoNnSRXzKuNqXMTDrnDs=.sha256

```
~/.npm-global/lib/node_modules/scuttlebot/node_modules/secure-scuttlebutt/minimal.js:41
            msg.unbox = key.toString('base64')
                            ^
TypeError: Cannot read property 'toString' of undefined
    at unbox (~/.npm-global/lib/node_modules/scuttlebot/node_modules/secure-scuttlebutt/minimal.js:41:29)
    at Object.decode (~/.npm-global/lib/node_modules/scuttlebot/node_modules/secure-scuttlebutt/minimal.js:88:14)
    at ~/.npm-global/lib/node_modules/scuttlebot/node_modules/flumelog-offset/inject.js:45:22
    at ~/.npm-global/lib/node_modules/scuttlebot/node_modules/flumelog-offset/frame/recoverable.js:48:11
    at ~/.npm-global/lib/node_modules/scuttlebot/node_modules/aligned-block-file/blocks.js:74:11
    at get (~/.npm-global/lib/node_modules/scuttlebot/node_modules/aligned-block-file/blocks.js:29:7)
    at next (~/.npm-global/lib/node_modules/scuttlebot/node_modules/aligned-block-file/blocks.js:53:7)
    at ~/.npm-global/lib/node_modules/scuttlebot/node_modules/aligned-block-file/blocks.js:63:25
    at get (~/.npm-global/lib/node_modules/scuttlebot/node_modules/aligned-block-file/blocks.js:29:7)
    at next (~/.npm-global/lib/node_modules/scuttlebot/node_modules/aligned-block-file/blocks.js:53:7)
```

### PR DESCRIPTION
**Pull request by christianbundy**
Resolves #237 

This diff is just an indentation fix plus:

```diff
-msg.unbox = key.toString('base64')
+if (key) {
+  msg.unbox = key.toString('base64')
+}
```

I'm unclear on *how* this could be happening, but it seems that the following must be true to trigger this error:

- `isFunction(unbox) === false`
- `key == null`
- `unbox.key == null`

@dominictarr @ahdinosaur Is this fine, or should we throw an error if we can't set `msg.value.unbox`?

### CODE REVIEW

**Comment by dominictarr** [comment]
this could happen if someone has an old unboxer that doesn't do the key separately, this could happen.

I also see some other problems with this code - the top `if(isFunction...) else ... ` should all be inside `if(key) {...}` because if a key is provided, it should be used for decryption explicitly.

File: minimal.js [hunk 1]
```diff
@@ -20,16 +20,15 @@ function unbox(data, unboxers, key) {
   var plaintext
   if(data && isString(data.value.content)) {
     for(var i = 0;i < unboxers.length;i++) {
-      var unbox = unboxers[i]
-      if(isFunction(unbox)) {
-        plaintext = unbox(data.value.content, data.value)
+      var unboxer = unboxers[i]
+
+      if (key) {
+        plaintext = unboxer.value(data.value.content, key)
```
Comment (christianbundy) [inline minimal.js]: @dominictarr Like this?

Comment (dominictarr) [reply minimal.js]: almost...

**Comment by christianbundy** [comment]
Oh, that makes sense. Thanks!

**Pull request merged**